src/platform/silabs/platformAbstraction/WiseMcuSpam.cpp

```diff
@@ -143,21 +143,26 @@ void sl_button_on_change(uint8_t btn, uint8_t btnAction)
     // Currently the btn0 is pull-up resistor due to which is sends a release event on every wakeup
     if (btn == SL_BUTTON_BTN0_NUMBER)
     {
-        if (btnAction == BUTTON_PRESSED)
+        if ((btnAction == BUTTON_PRESSED) && (sl_btn0_pressed == false))
         {
-            btn0_pressed = true;
+            sl_btn0_pressed = true;
         }
-        else if ((btnAction == BUTTON_RELEASED) && (btn0_pressed == false))
+        else if ((btnAction == BUTTON_RELEASED) && (sl_btn0_pressed == true))
         {
-            // if the btn was not pressed and only a release event came, ignore it
+            sl_btn0_pressed = false;
+        }
+        else if ((btnAction == BUTTON_PRESSED) && (sl_btn0_pressed == true))
+        {
+            // if the btn was already pressed and another press event came, ignore it
             return;
         }
-        else if ((btnAction == BUTTON_RELEASED) && (btn0_pressed == true))
+        else if ((btnAction == BUTTON_RELEASED) && (sl_btn0_pressed == false))
         {
-            btn0_pressed = false;
+            // if the btn was not pressed and only a release event came, ignore it
+            return;
```

Comment:
These conditions should be added at the start of the function with the `VerifyOrReturn` structure.

Reply:
@rosahay-silabs This marked resolved but was not addressed nor commented.